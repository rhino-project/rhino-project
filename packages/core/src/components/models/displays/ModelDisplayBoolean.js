import DisplayBoolean from 'rhino/components/forms/displays/DisplayBoolean';
import { useGlobalComponentForAttribute } from 'rhino/hooks/overrides';

// eslint-disable-next-line no-unused-vars
export const ModelDisplayBooleanBase = ({ model, ...props }) => (
  <DisplayBoolean {...props} />
);

const ModelDisplayBoolean = (props) =>
  useGlobalComponentForAttribute(
    'ModelDisplayBoolean',
    ModelDisplayBooleanBase,
    props
  );

export default ModelDisplayBoolean;
