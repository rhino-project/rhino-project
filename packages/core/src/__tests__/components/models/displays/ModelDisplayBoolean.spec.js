import ModelDisplayBoolean from 'rhino/components/models/displays/ModelDisplayBoolean';
import { sharedDisplayTests } from './sharedDisplayTests';

describe('ModelDisplayBoolean', () => {
  sharedDisplayTests(ModelDisplayBoolean);
});
