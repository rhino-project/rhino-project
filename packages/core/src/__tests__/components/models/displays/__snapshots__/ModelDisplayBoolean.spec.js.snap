// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModelDisplayBoolean should render the empty text when value is nullish 1`] = `
<DocumentFragment>
  <div
    class="custom-checkbox custom-control"
  >
    <input
      aria-invalid="false"
      class="custom-control-input"
      disabled=""
      id="dummy"
      name="dummy"
      path="dummy"
      readonly=""
      type="checkbox"
    />
    <label
      class="custom-control-label"
      for="dummy"
    />
  </div>
</DocumentFragment>
`;

exports[`ModelDisplayBoolean should render the overridden empty text when value is nullish 1`] = `
<DocumentFragment>
  <div
    class="custom-checkbox custom-control"
  >
    <input
      aria-invalid="false"
      class="custom-control-input"
      disabled=""
      empty="baz"
      id="dummy"
      name="dummy"
      path="dummy"
      readonly=""
      type="checkbox"
    />
    <label
      class="custom-control-label"
      for="dummy"
    />
  </div>
</DocumentFragment>
`;

exports[`ModelDisplayBoolean should render with global override shorthand 1`] = `
<DocumentFragment>
  <div>
    Bar
  </div>
</DocumentFragment>
`;

exports[`ModelDisplayBoolean should render with global override shorthand for model 1`] = `
<DocumentFragment>
  <div>
    Bar
  </div>
</DocumentFragment>
`;

exports[`ModelDisplayBoolean should render with global override shorthand for model and attribute 1`] = `
<DocumentFragment>
  <div>
    Bar
  </div>
</DocumentFragment>
`;
